### ISSUE TITLE
Top CMakeLists.txt assumes debug configuration if CMAKE_BUILD_TYPE is not Release

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#287]
```
Currently, CMakeLists.txt thinks that if the build type is not Release, then 
debug configuration should be used:


if(CMAKE_BUILD_TYPE MATCHES RELEASE OR CMAKE_BUILD_TYPE MATCHES "Release")
	message(STATUS "Building Release")
	set(Optz true)
else()
	message(STATUS "Building Debug")
	set(Optz false)
endif()


But this doesn't account for custom build types, like the one used in Gentoo 
(see 
http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/eclass/cmake-utils.eclas
s?view=markup for details on CMake project handling). The relevant code could 
be changed to


if(CMAKE_BUILD_TYPE MATCHES DEBUG OR CMAKE_BUILD_TYPE MATCHES "Debug")
	message(STATUS "Building Debug")
	set(Optz false)
else()
	message(STATUS "Building Release")
	set(Optz true)
endif()
```

-----
Original issue reported on code.google.com by `andrey.vihrov` on 24 Sep 2011 at 9:03

**Comment by GoogleCodeExporter** [comment]
```
fixed in r6040:0fbb2689cf48, thanks for reporting
```

-----
Original comment by `vittorio...@gmail.com` on 30 Sep 2011 at 8:43
* Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
CMake file changed
```

-----
Original comment by `jerryma1...@gmail.com` on 21 Dec 2011 at 1:00
* Changed state: **Verified**
